src/GameState/index.tsx

```diff
@@ -96,10 +97,34 @@ export default function GameState({ navigation, route, player }: any) {
     navigation.goBack();
   };
 
+  const initWebSocket = () => {
+    WS = new WebSocket(GATEWAY_URL);
+    WS.onopen = () => {
+      WS.send(JSON.stringify({ event: 'initGame', data: { gameId } }));
+    };
+
+    WS.onmessage = (e: any) => {
+      const gameState = JSON.parse(e.data)?.gameState as Game;
+      dispatch({ type: 'game', value: gameState });
+    };
+
+    WS.onerror = (e: any) => {
+      const toast = useToast();
+      toast.show({
+        title: 'Error',
+        status: 'error',
+        description: 'An error occurred with the live function',
+      });
+    };
+  };
+
   useEffect(() => {
     getGame();
+    initWebSocket();
```

Comment:
To close your websocket, you must do it as return of your useEffect.

```javascript
useEffect(() => {
    getGame();
    initWebSocket();
    return () => {
        if (WS) WS.close();
     };
}, []);
```

https://fr.reactjs.org/docs/hooks-reference.html#cleaning-up-an-effect